src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -138,5 +127,28 @@ public string GetSearchParameterHashForResourceType(string resourceType)
 
             return null;
         }
+
+        public void AddNewSearchParameters(IList<ITypedElement> searchParameters)
+        {
+            SearchParameterDefinitionBuilder.Build(
+                searchParameters,
+                UrlLookup,
+                TypeLookup,
+                _modelInfoProvider);
+
+            CalculateSearchParameterHash();
+        }
+
+        private void CalculateSearchParameterHash()
```

Comment:
>CalculateSearchParameterHash [](start = 21, length = 28)

I would add filter to this function. Calculating all hashes during start up is right thing to do, but recalculate them all again if only we update one search parameter is sound a bit overkill.

Reply:
This is a good idea.  After a quick look, we need to do some additional refactoring to get the necessary info to be able to filter correctly.  The Build method needs to return a list of affected resource types.  I think we should split this to a separate work item, as this PR is big enough already :).  Added item #78077.